rightscale/rsc/client.go

```diff
@@ -722,3 +735,23 @@ func (inFields Fields) onlyPopulated() Fields {
 	}
 	return outFields
 }
+
+// analyzeSource checks that the defition of the source is valid, returning an error if it's not
+// If the definition is valid, expectsOuputs boolean indicates if the defition includes the "return" keyword,
+// which indicates that output values are expected.
+func analyzeSource(source string) (expectsOutputs bool, err error) {
+	const validDefinition = "^[[:blank:]\n]*define[[:blank:]]*[\\w_\\.]+[[:blank:]]*\\([@$\\w _,]*\\)[[:blank:]]+(return.*[[:blank:]]+)?do"
+
+	r := regexp.MustCompile(validDefinition)
```

Comment:
The above could be a global variable so the regex doesn't get recompiled for each invocation (probably doesn't change much in reality).